### PR TITLE
[#department-of-veterans-affairs/va-virtual-agent#1102] rename skill …

### PR DESCRIPTION
**Pull request by nichelous-herndon**
## Summary
Added event based prescription generic entry and exit. Virtual Agent Chatbot Team

## Related issue(s)
department-of-veterans-affairs/va-virtual-agent#1102

## Testing done
Used to use message based trigger.
Create event, trigger event, check for event to go to and exit prescriptions skill. Also tested null event scenario.

## Screenshots
N/A

## What areas of the site does it impact?
Chatbot prescription skill

## Acceptance criteria
Create event, trigger event, check for event to go to and exit prescriptions skill. Also tested null event scenario.
The session storage is set on event.

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [N/A] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:
N/A

## Requested Feedback
N/A

### CODE REVIEW

File: src/applications/virtual-agent/components/webchat/helpers/startConvoAndTrackUtterancesHelpers.js [hunk 1]
```diff
@@ -104,9 +104,9 @@ export const processIncomingActivity = ({ action, dispatch }) => () => {
   const eventName = action?.payload?.activity?.name ?? '';
 
   // use event name for rxSkill
-  const rxSkillWasTriggered = eventName === 'RX_Skill_Entry';
+  const rxSkillWasTriggered = eventName === 'Skill_Entry';
```
Comment (ksk385) [inline src/applications/virtual-agent/components/webchat/helpers/startConvoAndTrackUtterancesHelpers.js:107]: We should confirm that the Skill Entered is RX_Skill. We would need to check for the value as well. @nichelous-herndon

File: src/applications/virtual-agent/tests/components/webchat/startConvoAndTrackUtterances.unit.spec.js [hunk 2]
```diff
@@ -283,7 +283,7 @@ describe('makeBotStartConvoAndTrackUtterances actions', () => {
       const spyDispatchEvent = sandbox.spy(window, 'dispatchEvent');
       const activity = {
         type: 'event',
-        name: 'RX_Skill_Exit',
+        name: 'Skill_Exit',
```
Comment (ksk385) [inline src/applications/virtual-agent/tests/components/webchat/startConvoAndTrackUtterances.unit.spec.js:286]: We need to mock the value as well. @nichelous-herndon

File: src/applications/virtual-agent/components/webchat/helpers/startConvoAndTrackUtterancesHelpers.js (hunk 1 above)
Comment (nichelous-herndon) [reply src/applications/virtual-agent/components/webchat/helpers/startConvoAndTrackUtterancesHelpers.js:107]: good idea. I'll add the changes.

Comment (nichelous-herndon) [reply src/applications/virtual-agent/components/webchat/helpers/startConvoAndTrackUtterancesHelpers.js:108]: @ksk385 Added the changes.

File: src/applications/virtual-agent/tests/components/webchat/startConvoAndTrackUtterances.unit.spec.js (hunk 2 above)
Comment (nichelous-herndon) [reply src/applications/virtual-agent/tests/components/webchat/startConvoAndTrackUtterances.unit.spec.js:287]: @ksk385 Added the changes.

File: src/applications/virtual-agent/components/webchat/helpers/startConvoAndTrackUtterancesHelpers.js [hunk 3]
```diff
@@ -102,16 +102,19 @@ export const processIncomingActivity = ({ action, dispatch }) => () => {
   }
 
   const eventName = action?.payload?.activity?.name ?? '';
+  const eventValue = action?.payload?.activity?.value ?? '';
 
   // use event name for rxSkill
-  const rxSkillWasTriggered = eventName === 'RX_Skill_Entry';
+  const rxSkillWasTriggered = eventName === 'Skill_Entry';
```
Comment (ksk385) [inline src/applications/virtual-agent/components/webchat/helpers/startConvoAndTrackUtterancesHelpers.js:108]: I would make these names generics as well now @nichelous-herndon

File: src/applications/virtual-agent/components/webchat/helpers/startConvoAndTrackUtterancesHelpers.js [hunk 4]
```diff
@@ -102,16 +102,19 @@ export const processIncomingActivity = ({ action, dispatch }) => () => {
...
   // use event name for rxSkillExit
-  const rxSkillWasExited = eventName === 'RX_Skill_Exit';
-  if (rxSkillWasTriggered) {
+  const rxSkillWasExited = eventName === 'Skill_Exit';
```
Comment (ksk385) [inline src/applications/virtual-agent/components/webchat/helpers/startConvoAndTrackUtterancesHelpers.js:110]: `skillWasExited` @nichelous-herndon

Comment (nichelous-herndon) [reply src/applications/virtual-agent/components/webchat/helpers/startConvoAndTrackUtterancesHelpers.js]: changes made @ksk385

**Review by karankrishnani — APPROVED** [review]
Looks good

**Pull request merged**